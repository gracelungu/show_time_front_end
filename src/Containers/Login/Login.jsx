import React, { Component } from "react";
import { Link } from 'react-router-dom';
import "./Login.scss";

import google from "../../assets/images/google.png";
import facebook from "../../assets/images/facebook.png";

class Login extends Component {
    render() {
        return (
            <section className="login-page flex justify-center align-center">
                <div className="login ">
                    <div className="top flex justify-between align-start">
                        <div>
                            <div className="title">Login</div>
                            <div className="subtitle">Welcome again</div>
                        </div>
                        <div className="social flex align-center">
                            <img src={google} width="80px" height="80px" alt="" />
                            <img src={facebook} width="80px" height="80px" alt="" />
                        </div>
                    </div>
                    <form action="" class="flex justify-between">
                        <div className="section">
                            <div className="field">
                                <div className="label">Email</div>
                                <input type="email" placeholder="" />
                            </div>
                            <div className="field">
                                <div className="label">Password</div>
                                <input type="password" />
                            </div>
                            <div className="btn primary text-white">
                                LOGIN
                            </div>

                            <div className="other">
                                <span>Don't have an account ? </span> <Link to="/signup"><b>Register</b></Link>
                            </div>

                        </div>
                    </form>
                </div>
            </section >
        );
    }
}

export default Login;
