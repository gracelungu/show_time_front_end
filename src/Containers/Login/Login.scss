@import "../../variables.scss";

.login-page{
    width: 100%;
    height: 100vh;
    color:#fff;
    .login{
        width: 26%;
        .top{
            .title{
                font-size: 1.6em;
                padding-bottom:10px;
            }
            .subtitle{
                color:#707070;
                font-size: .9em;
            }
            padding-bottom: 60px;
        }
        form{
            width:100%;
            .section{
                width: 100%;
                .field{
                    width: 90%;
                    margin-bottom:50px;
                    .label{
                        font-size: .9em;
                        padding-bottom: 30px;
                    }
                    input{
                        width: 100%;
                        padding:12px 15px;
                        font-size: .9em;
                        color:#ccc;
                        background: #171717;
                        border: 1px solid #2F2F2F;
                    }
                }
                .other{
                    text-align: right;
                    padding-top: 30px;
                    font-size: .9em;
                }
            }
        }
    }
}

@media only screen and (max-width: 768px) {
    .login-page{
        .login{
            width: 90%;
            margin-top: 0px;
            margin-bottom: 40px;
            form{
                flex-direction: column;
                .section{
                    width: 100%;
                }
            }
        }
    }
}