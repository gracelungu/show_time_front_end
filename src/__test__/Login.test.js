import React from "react";
import thunk from "redux-thunk";
import { mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import Login from "../Containers/Login/Login";

let wrapper, store;
const props = {};
const mockStore = configureMockStore([thunk]);
const storeStateMock = {
  auth: { loading: false },
  error: null
};

describe("Login.jsx", () => {
  beforeEach(() => {
    store = mockStore(storeStateMock);
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <Login {...props} />
        </Router>
      </Provider>
    );
  });

  test("should render Login.jx", () => {
    wrapper.find(".text-white").simulate("click");
    expect(wrapper.find(".login-page").length).toBe(1);
  });
});
