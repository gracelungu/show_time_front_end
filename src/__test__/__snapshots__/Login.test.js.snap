// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.jsx should render Home.jx 1`] = `ShallowWrapper {}`;

exports[`Login.jsx should render Login.jx 1`] = `undefined`;
